app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,634 @@
+    const [customListData, setCustomListData] = useState<DataType | UserProfileSection[]>([]);
+
+    // Callbacks
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults([]);
+    }, []);
+
+    // This is the button to submit multiselect options
+    const rightButton = useMemo(() => {
+        const base = buildNavigationButton(
+            SUBMIT_BUTTON_ID,
+            'integration_selector.multiselect.submit.button',
+            undefined,
+            intl.formatMessage({id: 'integration_selector.multiselect.submit', defaultMessage: 'Done'}),
+        );
+        base.enabled = true;
+        base.showAsAction = 'always';
+        base.color = theme.sidebarHeaderTextColor;
+        return base;
+    }, [theme.sidebarHeaderTextColor, intl]);
+
+    const handleSelectItem = useCallback((item: Selection) => {
+        if (!isMultiselect) {
+            handleSelect(item);
+            close();
+            return;
+        }
+
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS: {
+                const currentSelected = multiselectSelected as Dictionary<UserProfile>;
+                const typedItem = item as UserProfile;
+                const multiselectSelectedItems = {...currentSelected};
+
+                if (currentSelected[typedItem.id]) {
+                    delete multiselectSelectedItems[typedItem.id];
+                } else {
+                    multiselectSelectedItems[typedItem.id] = typedItem;
+                }
+
+                setMultiselectSelected(multiselectSelectedItems);
+                break;
+            }
+            case ViewConstants.DATA_SOURCE_CHANNELS: {
+                const currentSelected = multiselectSelected as Dictionary<Channel>;
+                const typedItem = item as Channel;
+                const multiselectSelectedItems = {...currentSelected};
+                if (currentSelected[typedItem.id]) {
+                    delete multiselectSelectedItems[typedItem.id];
+                } else {
+                    multiselectSelectedItems[typedItem.id] = typedItem;
+                }
+
+                setMultiselectSelected(multiselectSelectedItems);
+                break;
+            }
+            default: {
+                const currentSelected = multiselectSelected as Dictionary<DialogOption>;
+                const typedItem = item as DialogOption;
+                const multiselectSelectedItems = {...currentSelected};
+                if (currentSelected[typedItem.value]) {
+                    delete multiselectSelectedItems[typedItem.value];
+                } else {
+                    multiselectSelectedItems[typedItem.value] = typedItem;
+                }
+                setMultiselectSelected(multiselectSelectedItems);
+            }
+        }
+    }, [integrationData, multiselectSelected, isMultiselect, dataSource, close, handleSelect]);
```

Comment:
close is constant between renders (it is a function outside of the component) so no need to add it to the dependencies.